src/server/frontend_wayland/xdg_activation_v1.cpp

```diff
@@ -309,10 +326,9 @@ void mf::XdgActivationTokenV1::commit()
         }
 
         auto const& scene_surface = scene_surface_opt.value();
-
         if (focus_controller->focused_surface() != scene_surface)
         {
-            mir::log_error("XdgActivationTokenV1::commit the focused surface is not the surface that requesteda ctivation");
+            mir::log_error("XdgActivationTokenV1::commit the focused surface is not the surface that requested activation");
```

Comment:
The more I think about this, the more I wonder if a correct behavior would be to check this when we receive the `activate` request. E.g.: Surface A commits its info and gets a token, then it passes the token to Surface B to activate. If Surface A is no longer focused (because the user selected a different surface in the time that it took for Surface B to come up), then we don't activate Surface B because it could be jarring for the user. WDYT?

Reply:
Although maybe they should be warnings instead of errors. But that would be a more appropriate time to them IMO

Reply:
Yeah unfortunately there doesn't exist an error code for that, which I think is a bit silly

Reply:
> If Surface A is no longer focused

It's not about not being focused (because, in theory, you could've switched to C and back to A) - it's about whether there was a focus _change_ between issuing the token and receiving the activation request.

Reply:
I believe it's really that simple:

1. issue the token
1. invalidate the token on key/button down or focus change (say A went away)
1. respect the activation request if token is valid

Reply:
I think instead of disrespecting when _any_ other surface is focused is a bad idea. The protocol allows users to specify _any_ surface or even an app_id for a number of surfaces. Because of this, it may be valid for clients to:
1. Set the "requesting surface" to Surface B
2. Request the token on Surface A
3. Focus Surface B
4. Create and Activate Surface C (expecting Surface B to be focused)

It is contrived, but by sending up the surface over the protocol, I would only expect that particular surface to matter.

Reply:
I agree about the key event thing though :)